config/defaults_template.yaml

```diff
@@ -31,3 +31,6 @@ languages:
     buildImage: l.gcr.io/google/bazel:latest
     runImage: "{{ .ImagePrefix }}python:{{ .Version }}"
 
+  - language: csharp
+    buildImage: "{{ .InitImagePrefix }}build/csharp:{{ .Version }}"
```

Comment:
We may want to create a new variable, `.BuildImagePrefix` which is the `.InitImagePrefix` + `"build/"` or some override.

The problem is that the image "build/csharp" may not be valid for all registries. DockerHub, for instance, would not likely work. With the other examples, the images are valid even if the `.InitImagePrefix` or `.ImagePrefix` are not set. 

We don't have to do it in this PR, since we know we are only using gcr at the moment. But, we should do it... since we have avoided most vendor lock-in thus far.